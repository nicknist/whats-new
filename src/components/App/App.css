* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

#blockColorblindContent {
  display: none;
}

.app {
  display: grid;
  grid-template-areas:
            "head head"
            "nav main";
  grid-template-rows: 1fr 9fr;
  grid-template-columns: 1fr 4fr;
  background: #141527;
  height: 100vh;
  text-align: center;
  width: 100%;
}

.menu {
  grid-area: nav;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

header {
  grid-area: head;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.news-container {
  grid-area: main;
}

/* some sort of grid set up */
