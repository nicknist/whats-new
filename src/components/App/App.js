import React, { Component } from 'react';
import './App.css';
import Menu from '../Menu/Menu'
import SearchForm from '../SearchForm/SearchForm'
import NewsContainer from '../NewsContainer/NewsContainer'

class App extends Component {
  constructor() {
    super();
    this.state = {
      search: '',
      currentPage: 'local',
      fullData: {
        local: [],
      },
      error: null,
      isLoading: false,
    }
  }

  componentDidMount() {
    this.setState({ isLoading: true })

    fetch('https://whats-new-api.herokuapp.com/api/v1/news')
      .then(response => response.json())
      .then(data => {
        this.setState({ fullData: data, isLoading: false })
      })
      .catch(error => this.setState({ error, isLoading: false}))
  }

  changePage = (event) => {
    this.setState({
      currentPage: event.target.id
     })
  }

  searchArticle = (input) => {
    const { fullData } = this.state;
    const searchedArticles = this.state.fullData[this.state.currentPage].filter(data =>  data.description.toUpperCase().includes(input.toUpperCase()));

    this.setState({
      currentPage: 'filtered',
      fullData: {
        ...fullData,
        filtered: searchedArticles
      }
    });
  }

  changeInput = (event) => {
    this.setState({ search: event.target.value });
  }

  render () {
    if (this.state.isLoading) {
      return <div className="loading-ring app-load"></div>
    }

    if (this.state.error) {
      return <p>There was an error loading the Page</p>
    }

    return (
      <div className="app">
        <Menu changePage={this.changePage} />
        <SearchForm searchArticle={this.searchArticle} changeInput={this.changeInput} currentInput={this.state.search}/>
        <NewsContainer data={this.state.fullData[this.state.currentPage]} />
      </div>
    );
  }
}

export default App;
