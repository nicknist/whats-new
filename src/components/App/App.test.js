import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  it('should display the loading screen page', () => {
    const wrapper = shallow(<App /> );
    wrapper.setState({ isLoading: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('should display an error message when error happens', () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ isLoading: false, error: true });

    expect(wrapper).toMatchSnapshot();
  });

  it('should display main page when not loading and no error', () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ error: false, isLoading: false });

    expect(wrapper).toMatchSnapshot();
  })

})
