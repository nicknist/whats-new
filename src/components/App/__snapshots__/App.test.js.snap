// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should display an error message when error happens 1`] = `
<p>
  There was an error loading the Page
</p>
`;

exports[`App should display main page when not loading and no error 1`] = `
<div
  className="app"
>
  <Menu
    changePage={[Function]}
  />
  <SearchForm
    searchArticle={[Function]}
  />
  <NewsContainer
    data={Array []}
  />
</div>
`;

exports[`App should display the loading screen page 1`] = `
<div
  className="loading-ring app-load"
/>
`;
